xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/dashboard/dashboard.js

```diff
@@ -375,24 +375,35 @@ XWiki.Dashboard = Class.create( {
   onEditGadgetClick : function(event) {
     var gadget = event.element().up('.gadget');
 
+    /**
+     * Finds the direct child element of the given element with the specified class name.
```

Comment:
Maybe it would be worth indicating that this always returns the first matching element.